spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -131,6 +142,16 @@
     this.splitLookback = options.get("lookback").map(Integer::parseInt).orElse(null);
     this.splitOpenFileCost = options.get("file-open-cost").map(Long::parseLong).orElse(null);
 
+    String scheme = "no_exist";
+    try {
+      FileSystem fs = FileSystem.get(SparkSession.active().sparkContext().hadoopConfiguration());
+      scheme = fs.getScheme().toLowerCase(Locale.ENGLISH);
+    } catch (IOException ioe) {
+      LOG.warn("Failed to get Hadoop Filesystem", ioe);
```

Comment:
Does it make sense to use the method `org.apache.hadoop.fs.FileSystem#getDefaultUri` instead? That does not throw an exception and does not need to get a `FileSystem` class. It gets the information from the `DEFAULT_URI`

Reply:
Actually, there's no guarantee that the table uses the default file system. I think this should check the file system of the table's `location` instead.

Reply:
That makes sense!

Reply:
Sure, let me change it.